File: Practica 4/Ejercicio 3/potencia.c
#include <stdio.h>

int potencia(int bas, int exp) {
    int res;
    res = bas *= exp;
    return res;
}

int main() {

    int bas, exp;

    printf("\nIntroduce la base: ");
    scanf("%d", &bas);

    printf("\nIntroduce el exponente: ");
    scanf("%d", &exp);

    while (bas < 0 || exp < 0) {
        printf("\nError, números nno válidos");
        printf("\nVuelva a introducir la base: ");
        scanf("%d", &bas);
        printf("\nVuelve a introducir el exponente: ");
        scanf("%d", &exp);
    }

    int resultado = potencia(bas, exp);
    printf("\nEl resultado de elevar %d a %d es: %d", bas, exp, resultado);

    return 0;
}

File: Practicas/Practica 6/Ejercicio 3/potencia.py
def potencia(bas, exp):
    res = bas ** exp
    print("\nEl resultado de elevar",bas,"al exponente",exp,"es:",res)

def main():
    
    bas = (int(input("\nIntroduzca la base: ")))
    exp = (int(input("\nIntroduce el exponente: ")))
    
    while (bas < 0 or exp < 0):
        print("\nError, introduzca una base y exponente positivos")
        bas= int(input("\nIntroduzca uns base positiva: "))
        exp = int(input("\nIntroduzca un exponente positivo: "))
        
    potencia(bas, exp)
    
if __name__ == "__main__":
    main()